core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -2703,6 +2711,26 @@ class ReplicaManagerTest {
       assertEquals(HostedPartition.None, replicaManager.getPartition(tp0))
       assertFalse(readRecoveryPointCheckpoint().contains(tp0))
       assertFalse(readLogStartOffsetCheckpoint().contains(tp0))
+
+      // verify that there is log to be deleted
+      val logsToBeDeleted = replicaManager.logManager.logsToBeDeleted
+      assertEquals(1, logsToBeDeleted.size())
+      val (removedLog, _) = logsToBeDeleted.take()
+      assertFalse(removedLog.logDirFailureChannel.hasOfflineLogDir(logDir.toString))
+
+      // trigger the background deletion
+      var kafkaStorageExceptionCaptured = false
+      try {
+        removedLog.delete()
+      } catch {
+        case _: KafkaStorageException =>
+          kafkaStorageExceptionCaptured = true
+      }
+      assertEquals(expectedExceptionInBackgroundDeletion, kafkaStorageExceptionCaptured)
```

Comment:
I wonder if this test is the appropriate place for this check because it does not really test the stop replica path. Would it make sense to have it in a separate test? @junrao What do you think?

Reply:
@dajac : Yes, perhaps we could have a separate log level test for this exception if not covered by existing tests.

Reply:
I've removed this section from this test and added a new log level test UnifiedLogTest.testRenamingDirWithoutReinitialization. Please take a look at the latest PR. Thank you!